cincinnati/src/plugins/internal/graph_builder/openshift_secondary_metadata_parser/plugin.rs

```diff
@@ -265,46 +265,88 @@ impl OpenshiftSecondaryMetadataParserPlugin {
-                    to.build
-                        .push(semver::Identifier::AlphaNumeric(arch.to_string()));
-                };
-
-                // find version in the graph
-                let release_id = match io.graph.find_by_version(&to.to_string()) {
-                    Some(release_id) => release_id,
-                    None => {
-                        warn!("Release with version {} not found in graph", to);
-                        return Ok(());
+                            // Default to blocking all architectsures
+                            architectures
+                                .iter()
+                                .map(|blocked_architecture| {
+                                    let mut to = to.clone();
+                                    to.build = blocked_architecture.clone();
+                                    to
+                                })
+                                .collect()
+                        }
                     }
                 };
 
-                // add metadata to block edge using the `previous.remove_regex` metadata
-                io.graph
-                    .get_metadata_as_ref_mut(&release_id)
-                    .context(format!(
-                        "[blocked_edges] Getting mutable metadata for {}",
-                        &to.to_string()
-                    ))?
-                    .insert(
-                        format!("{}.{}", self.settings.key_prefix, "previous.remove_regex"),
-                        blocked_edge.from.to_string(),
-                    );
+                // find all versions in the graph
+                target_versions.iter().for_each(|to| {
+                    match io.graph.find_by_version(&to.to_string()) {
+                        Some(release_id) => {
+                            // add metadata to block edge using the `previous.remove_regex` metadata
+                            match &mut io.graph.get_metadata_as_ref_mut(&release_id).context(
+                                format!(
+                                    "[blocked_edges] Getting mutable metadata for {} failed.",
+                                    &to.to_string()
+                                ),
+                            ) {
+                                Ok(metadata) => {
+                                    metadata.insert(
+                                        format!(
+                                            "{}.{}",
+                                            self.settings.key_prefix, "previous.remove_regex"
+                                        ),
+                                        blocked_edge.from.to_string(),
+                                    );
+                                }
+                                Err(e) => warn!("{}", e),
+                            };
+                        }
+                        None => {
+                            info!("Release with version {} not found in graph", to);
```

Comment:
I don't think we expect that to happen (`blocked_edges` should not mention invalid nodes). Let throw a warning here?

Reply:
This is expected because some blocked edges are given without an architecture, hence we look for the given version for all architectures which may not exist. I'm fine changing this to be warning anyway.